### ISSUE TITLE
about optimizationk

### ISSUE THREAD

**Issue by lyrachord** [sladra/DACs#1]
Hi, 
Years ago, I read the DACs paper. Today  I redo it.
in dacs.c optimizationk(), I found this

`uint * weight = (uint *) malloc(sizeof(uint)*(maxInt+1));`

If the maxInt huge, too many memories to eat.

**Comment by sladra** [comment]
Yes, true. The construction process has not been optimized in terms of space or time, as the code is focused on optimizing time/space of the processing of the sequence (when the compressed sequence has been already created).

I will take not of this issue for the future, to optimize also the construction process.

Thank you for the note.
